dallinger/experiment_server/experiment_server.py

```diff
@@ -95,21 +96,26 @@ def success_response(**data):
 
 
 def error_response(error_type="Internal server error",
-                   error_text=None,
+                   error_text='',
                    status=400,
-                   participant=None):
+                   participant=None,
+                   simple=False):
     """Return a generic server error response."""
-    traceback.print_exc()
+    if sys.exc_info()[0]:
+        db.logger.exception("Failure for request: {}".format(dict(request.args)))
```

Comment:
The `logging` docs in the stdlib suggest it's not supported to call `logging.exception` apart from inside a try/except block.

I think this should be:

    exc_info = sys.exc_info()
    if exc_info[0]:
        db.logger.exception("Failure for request: {}".format(dict(request.args)), exc_info=exc_info)